scanners/android/android.go

```diff
@@ -46,73 +48,108 @@ func (scanner *Scanner) DetectPlatform(searchDir string) (_ bool, err error) {
 
 	log.TInfof("Searching for android files")
 
-	scanner.ProjectRoots, err = walkMultipleFileGroups(searchDir, projectFiles, skipDirs)
+	projectRoots, err := walkMultipleFileGroups(searchDir, projectFiles, skipDirs)
 	if err != nil {
-		return false, fmt.Errorf("failed to search for build.gradle files, error: %s", err)
+		return false, nil, fmt.Errorf("failed to search for build.gradle files, error: %s", err)
 	}
 
-	log.TSuccessf("Platform detected")
-
+	log.TPrintf("%d android files detected", len(projectRoots))
 	for _, file := range projectFiles {
 		log.TPrintf("- %s", file)
 	}
-	countDetected := len(scanner.ProjectRoots)
-	log.TPrintf("%d android files detected", countDetected)
-	nonZeroCount := countDetected > 0
-	return nonZeroCount, err
+
+	if len(projectRoots) == 0 {
+		return false, nil, nil
+	}
+	log.TSuccessf("Platform detected")
```

Comment:
in the case of `parseProjects` error we still end up by not detecting the project type

Reply:
Detection result is not used in case of error, so removed bool return value completely.